File: backend/src/main/java/isd/group6/hanu/entity/Product.java
package isd.group6.hanu.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import isd.group6.hanu.common.UploadFile;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDateTime;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@Document(collection = "products")
public class Product {

    @Id
    private String id;

    private String name;

    private String brand;

    private String category;

    private String model;

    private String voltage;

    private String power;

    private double price;

    private Integer status;

    private List<String> description;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private List<UploadFile> files;

    private List<String> fileIds;
}

File: frontend/src/views/Customer/Detail/Model/Product.ts
export class ProductModel {
    id: string;
    name: string;
    brand: string;
    category: string;
    model: string;
    voltage: string;
    power: string;
    price: number;
    image: string[];
    status: number;
    description: string[];

    constructor(
        id: string,
        name: string,
        brand: string,
        category: string,
        model: string,
        voltage: string,
        power: string,
        price: number,
        image: string[],
        status: number,
        description: string[]
    ) {
        this.id = id;
        this.name = name;
        this.brand = brand;
        this.category = category;
        this.model = model;
        this.voltage = voltage;
        this.power = power;
        this.price = price;
        this.image = image;
        this.status = status;
        this.description = description;
    }
}
